SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -334,6 +334,8 @@ TArray<FWorkerComponentData> EntityFactory::CreatePartitionEntityComponents(cons
 	Components.Add(Metadata(FString::Format(TEXT("PartitionEntity:{0}"), { VirtualWorker })).CreateMetadataData());
 	Components.Add(InterestFactory->CreatePartitionInterest(LbStrategy, VirtualWorker, bDebugContextValid).CreateInterestData());
 	Components.Add(AuthorityDelegation(DelegationMap).CreateAuthorityDelegationData());
+	Components.Add(Persistence().CreatePersistenceData());
+	Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::PARTITION_SHADOW_COMPONENT_ID));
```

Comment:
@yumnuska @samiwh thoughts on adding the partition shadow component to the well known snapshot partition also? I didn't for now because it didn't seem necessary, but not hugely opinionated,

Reply:
I don't think it's strictly necessary, but IMO its less confusing to include it (otherwise it might seem like it was omitted by mistake?). I can also imagine it coming in handy at some point.

Reply:
one way it might come in handy is during a snapshot migration -- it means that we can assume that all partitions referred to by the AuthorityDelegation components will exist in the snapshot without having to special case EntityId == 4.

Reply:
all server partitions that is, which means we can assume other partitions are "client" partitions

Reply:
Cool - have added to the snapshot partition.